src/hooks/activityNavigation/ActivityProvider.tsx

```diff
@@ -19,11 +19,10 @@ export const ActivityProvider: FC<ActivityProviderProps> = ({
 }) => {
   const [currentActivityId, setCurrentActivityId] = useState<string>('')
 
-  useEffect(() => {
-    findNextActiveActivity()
-  }, [])
-
   const findCurrentActivity = (): Activity | undefined => {
+    if (activities.length === 0 || isNil(currentActivityId)) {
```

Comment:
currentActivityId will never be null or undefined, it's set as an empty string by default in the useState constructor above. i think you want to check that it's an empty string

`    if (activities.length === 0 || currentActivityId === '') {`

alternatively, initiate currenctActivityId as undefined